### ISSUE TITLE
Posting IM's directly to users is not working

### ISSUE THREAD

**Issue by chrisBrookes93** [pmxbot/pmxbot#101]
There is currently an issue in v1122.13.0 where it errors when attempting to send a message directly to a user (IM). 
Error:
```
::ffff:192.168.16.1 - - [31/Oct/2022:13:29:46] "POST /<slack handle> HTTP/1.1" 200 26 "" "python-requests/2.14.2"
Unhandled exception transmitting message: "<REDACTED>'"
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/pmxbot/core.py", line 432, in out
    sent = self.allow(channel, s) and self.transmit(channel, s)
  File "/usr/local/lib/python3.8/site-packages/pmxbot/slack.py", line 107, in transmit
    self.slack.web_client.chat_postMessage(
  File "/usr/local/lib/python3.8/site-packages/slack_sdk/web/client.py", line 2072, in chat_postMessage
    return self.api_call("chat.postMessage", json=kwargs)
  File "/usr/local/lib/python3.8/site-packages/slack_sdk/web/base_client.py", line 156, in api_call
    return self._sync_send(api_url=api_url, req_args=req_args)
  File "/usr/local/lib/python3.8/site-packages/slack_sdk/web/base_client.py", line 187, in _sync_send
    return self._urllib_api_call(
  File "/usr/local/lib/python3.8/site-packages/slack_sdk/web/base_client.py", line 309, in _urllib_api_call
    return SlackResponse(
  File "/usr/local/lib/python3.8/site-packages/slack_sdk/web/slack_response.py", line 199, in validate
    raise e.SlackApiError(message=msg, response=self)
slack_sdk.errors.SlackApiError: The request to the Slack API failed. (url: https://www.slack.com/api/chat.postMessage)
The server responded with: {'ok': False, 'error': 'invalid_arguments', 'response_metadata': {'messages': ['[ERROR] missing required field: channel']}}

```

According to the docs ([here](https://api.slack.com/methods/chat.postMessage#dm)) the user ID needs to be specified in the `channel` argument.  The fix is to resolve the user name to a user ID (if we fail to get a channel ID) and pass that in as the channel. 

The previous code that was mistakenly removed is here: https://github.com/pmxbot/pmxbot/blob/v1122.11.0/pmxbot/slack.py#L105-L127

As per the previous code, we should also support email addresses

**Comment by jaraco** [comment]
I presume you wish to work on this one. Let me know if you need assistance. Feel free to bring back slacker if needed.

**Comment by jaraco** [comment]
Fixed in #103.
